karyon3-eureka/src/main/java/com/netflix/karyon/eureka/EurekaModule.java

```diff
@@ -0,0 +1,43 @@
+package com.netflix.karyon.eureka;
+
+import com.google.inject.AbstractModule;
+import com.netflix.appinfo.EurekaInstanceConfig;
+import com.netflix.appinfo.HealthCheckHandler;
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.appinfo.providers.EurekaConfigBasedInstanceInfoProvider;
+import com.netflix.archaius.guice.ArchaiusModule;
+import com.netflix.discovery.DiscoveryClient;
+import com.netflix.discovery.EurekaClient;
+import com.netflix.discovery.EurekaClientConfig;
+
+/**
+ * Add this module to your project to enable Eureka client and registration
+ * 
+ * @author elandau
+ *
+ */
+public final class EurekaModule extends AbstractModule {
+    @Override
+    protected void configure() {
+    	install(new ArchaiusModule());
+    	
```

Comment:
ApplicationInfoManager is a concrete class and can be injected using just in time bindings.  I don't think it provides any value as an eager singleton.